Atomic/uninstall.py

```diff
@@ -36,34 +45,23 @@ def __init__(self):
         super(Uninstall, self).__init__()
 
     def uninstall(self):
-        if self.syscontainers.get_checkout(self.args.image):
-            return self.syscontainers.uninstall(self.args.image)
+        if self.args.debug:
+            util.write_out(str(self.args))
 
-        self.inspect = self._inspect_container()
-        if self.inspect and self.force:
-            self.force_delete_containers()
+        beu = BackendUtils()
         try:
-            # Attempt to remove container, if it exists just return
-            self.d.stop(self.name)
-            self.d.remove_container(self.name)
-        except NotFound:
-            # On exception attempt to remove image
-            pass
-
-        self.inspect = self._inspect_image()
-        if not self.inspect:
-            raise ValueError("Image '%s' is not installed" % self.image)
-
-        args = self._get_args("UNINSTALL")
-        if args:
-            cmd = self.gen_cmd(args + self.quote(self.args.args))
-            cmd = self.sub_env_strings(cmd)
-            self.display(cmd)
-            util.check_call(cmd, env=self.cmd_env())
+            be, img_obj = beu.get_backend_and_image_obj(self.args.image, str_preferred_backend=self.args.storage)
+        except ValueError as e:
+            if 'ostree' in [x().backend for x in beu.available_backends]:
```

Comment:
Because the logic for uninstall differs for the docker and ostree backends.  In the case of docker, we look for an image and then remove its associated containers.  In the case of ostree, we remove the container.   Therefore, for ostree, I do two things conditionally.  First I have to make sure that ostree is an available backend and then I see if it has the container.